packages/components/Spinner/src/Spinner.tsx

```diff
@@ -0,0 +1,72 @@
+/** @jsx withSlots */
+import { compose, UseSlots, buildProps, mergeProps, withSlots } from '@fluentui-react-native/framework';
+import { NativeModules, ViewProps } from 'react-native';
+import * as React from 'react';
+import { ensureNativeComponent } from '@fluentui-react-native/component-cache';
+
+const spinnerName = 'MSFSpinnerView';
+
+const NativeSpinnerView = ensureNativeComponent(spinnerName);
+
+export type Size = 'xSmall' | 'small' | 'medium' | 'large' | 'xLarge';
+
+interface ExportedConstants {
+  sizes: { [key in Size]: number };
+}
+
+const ExportedNativeConstants: ExportedConstants = NativeModules.MSFSpinnerViewManager;
+
+export type SpinnerTokens = {
+  /**
+   * Color of spinner
+   */
+  color?: string;
+};
+
+export type SpinnerProps = ViewProps & {
+  /**
+   * True if the spinner should stop animating when its not visible
+   */
+  hidesWhenStopped?: boolean;
+
+  /**
+   * True if animating (animating as soon as rendered by default)
+   */
+  animating?: boolean;
+
+  /**
+   * Supported spinner sizes (default small)
+   */
+  size?: Size;
+};
+
+export type NativeSpinnerViewProps = ViewProps & SpinnerProps & SpinnerTokens;
+
+interface SpinnerType {
+  props: SpinnerProps;
+  slotProps: { root: NativeSpinnerViewProps };
+  tokens: SpinnerTokens;
+}
+
+export const Spinner = compose<SpinnerType>({
+  displayName: spinnerName,
+  tokens: [
+    {
+      color: 'rgb(145,145,145)',
+    },
+    spinnerName,
+  ],
+  slotProps: {
+    root: buildProps((tokens) => ({ ...tokens }), []),
+  },
+  slots: { root: NativeSpinnerView },
+  render: (props: NativeSpinnerViewProps, useSlots: UseSlots<SpinnerType>) => {
+    const size = ExportedNativeConstants.sizes[props.size || 'small'];
+    const Root = useSlots(props).root;
+    return (rest: NativeSpinnerViewProps, children: React.ReactNode) => (
+      <Root {...mergeProps(props, rest)} style={[props.style, { height: size, width: size }]}>
```

Comment:
This is creating a new style object on every render which will then cause the produced style to be new every time.

Use the memo-cache package to get around this. Because the only variable is size, you only need to cache on the basis of size.  So the code would look something like this:
```tsx
// outside of the spinner component
import { getMemoCache } from '@fluentui-react-native/framework';
const cache = getMemoCache();

// inside render
const sizeProps = cache(() => ({ style: { height: size, width: size }}, [size]);

// root render then becomes:
<Root {...mergeProps(sizeProps, props, rest)>
```

Alternatively you could add the size value to the tokens then do the following:

```ts
// add size to tokens above
// add this inside compose:
tokensThatAreAlsoProps: ['size']
// then add it to the root buildProps call
slotProps: {
  root: buildProps(
    (tokens) => ({ color: tokens.color, style: { height: tokens.size, width: tokens.size } }), 
    ['color', 'size']
  )
```

This second approach does the same thing but allows customization of the default value for size.

Reply:
Size doesn't feel like it's worthy of being a token for me, as I feel different parts of an app will want to size their Spinner differently. Would it make sense to have tokens differ between different parts of an app, AKA the Spinner in this stack inherits token 1, while the spinner in this other stack inherits token 2?


I now remember when that memo cache was moved to 1.0.0, I'll look into using that. Thanks for the explanation :D

Reply:
I went with your second approach! I looked at documentation for the web shimmer that said "Don't use more than one spinner in a window" which then told me this lends itself to be a token.

Reply:
er, now it's correct. Sorry about that.